swap/swap.go

```diff
@@ -441,9 +479,11 @@ func (s *Swap) Balance(peer enode.ID) (int64, error) {
 func (s *Swap) Balances() (map[enode.ID]int64, error) {
 	balances := make(map[enode.ID]int64)
 
+	s.mapLock.RLock()
 	for peerID, peerBalance := range s.balances {
```

Comment:
i haven't found a way to iterate through this map in a way that could individually call `getBalance` each time.

i think it would be nice not to have references to the `balances` field (or the `mapLock`) outside of getters and setters-type of functions. 

do you think it's worth creating a function that encapsulates this block of code? (with the locks included)

Reply:
Two possible suggestions:
* Use `s.Lock`
* Implement a `getPeersWithBalance()` returning the list of peers (IDs) and then call individually `getBalance`

Reply:
sticking with the locks as they are.